packages/react/src/components/CodeSnippet/CodeSnippet.js

```diff
@@ -19,24 +19,36 @@ import getUniqueId from '../../tools/uniqueId';
 
 const { prefix } = settings;
 
-const rowHeightInPixels = 18;
-const defaultCollapsedHeightInRows = 15;
+const defaultMaxClosedNumberOfRows = 15;
+const defaultMaxExpandedNumberOfRows = 30;
+const defaultMinClosedNumberOfRows = 3;
+const defaultMinExpandedNumberOfRows = 16;
+const defaultShowLessText = 'Show less';
+const defaultShowMoreText = 'Show more';
+const defaultType = 'single';
+const defaultWrapText = false;
+
+const rowHeightInPixels = 16;
 
 function CodeSnippet({
-  className,
-  type,
+  ariaLabel,
   children,
+  className,
+  copyButtonDescription,
+  copyLabel, //TODO: Merge this prop to `ariaLabel` in `v11`
   disabled,
   feedback,
-  onClick,
-  ariaLabel,
-  copyLabel, //TODO: Merge this prop to `ariaLabel` in `v11`
-  copyButtonDescription,
-  light,
-  showMoreText,
-  showLessText,
   hideCopyButton,
-  wrapText,
+  light,
+  maxClosedNumberOfRows = defaultMaxClosedNumberOfRows,
+  maxExpandedNumberOfRows = defaultMaxExpandedNumberOfRows,
+  minClosedNumberOfRows = defaultMinClosedNumberOfRows,
+  minExpandedNumberOfRows = defaultMinExpandedNumberOfRows,
+  onClick,
+  showLessText = defaultShowLessText,
+  showMoreText = defaultShowMoreText,
+  type = defaultType,
+  wrapText = defaultWrapText,
```

Comment:
From @dakahn 

Tested NVDA with Firefox and JAWS with Edge/Chrome on Windows 10
A couple of notes:

In JAWS latest and NVDA the Show More button's label is being read twice as "Show more, show more button", could be some unneeded aria causing this.
In Edge/Chrome the scrolling content in not keyboard accessible
the horizontal scrolling is a violation of WCAG 1.4.10. The code snippet should reflow down the page and not require scrolling in two dimensions.

Reply:
as for things like `wrapText` -- as long as the out-of-the-box experience meets WCAG requirements we can document the problems with props that cause violations and we're good as far as v10 is concerned. We can look into deprecating `wrapText` in future versions though for sure.

Reply:
You are right, when hidden there is no scrollbar - but you can still scroll - which is one of the problems I was trying to address.
You can, In prod, end up with two way scrolling; if wrapText=false and the CodeSnippet has enough rows you do get two scrollbars.
One is on the CodeSnippet and one is on the container the snippet is in.
I made the issue much easier to encounter by enabling vertical scrolling and always showing the horizontal scrolling.

The existing issues, and my changes effect any OS, not just windows.

https://user-images.githubusercontent.com/5973642/110527396-e783b780-80e4-11eb-969a-1dbb5020b05d.mov

Don't get me wrong, I understand my solution is not 1.4.10 compliant.
I just think to be 1.4.10 compliant more discussion is needed on the points I made earlier.

Reply:
Actually, as the issue is simply an issue with two scrollbars.
So isn't the solution - at least for horizontal LTR languages to disallow wrapText?
This would allow only the vertical scrollbar and therefor be 1.4.10 compliant.

We still have to deal with...
- horizontal RTL latin languages - which should start right justified
- vertical languages - which should only have a vertical scroll bar
- and a proper wrapText to ensure we are breaking in correct places